data/src/main/java/com/example/pinterbest/data/repository/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,87 @@
+package com.example.pinterbest.data.repository
+
+import android.content.SharedPreferences
+import com.example.pinterbest.data.api.ApiClient
+import com.example.pinterbest.domain.common.ErrorMessage
+import com.example.pinterbest.domain.common.ResponseWithCookie
+import com.example.pinterbest.domain.common.Result
+import com.example.pinterbest.domain.entities.UserLogIn
+import com.example.pinterbest.domain.entities.UserSignUp
+import com.example.pinterbest.domain.repositories.AuthRepository
+import java.net.UnknownHostException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+
+class AuthRepositoryImpl(val preferences: SharedPreferences) : AuthRepository {
+    override suspend fun getCheckAuth(): Flow<Result<Int>> = flow {
+        emit(Result.Loading)
+        try {
+            val response = ApiClient().getInstance().getClient().getAuthCheck(
+                SessionRepositoryImpl(preferences).authProvider() ?: ""
```

Comment:
SessionRepositoryImpl must be provided outside of class via constructor. Creating any dependencies inside it's a bad practice

Reply:
Fixed